skema/program_analysis/TS2CAST/node_helper.py

```diff
@@ -0,0 +1,102 @@
+from skema.program_analysis.CAST2FN.model.cast import (
+   SourceRef
+)
+class NodeHelper(object):
```

Comment:
General question, just to consider: My understanding of the intent here in creating the NodeHelper is to walk the tree-sitter nodes and then add metadata that makes it easy to define a set of predicates (functions that return True/False to indicate whether a node satisfies some properties) as `is_*`, or to retrieve some information. If I'm following, your current approach create is to walk the tree-sitter tree and build up the dict while bookkeeping data that then supports the predicates and info retrieval. I'm wondering whether the NodeHelper bookkeeping needs to be a separate data structure or if the predicates and info could just be implemented to test/retrieve while walking the tree-sitter tree itself. I'm definitely not saying this is wrong, I'm just not sure I understand why there needs to be a separate creation of a persistent NodeHelper bookkeeping store.